src/Tutor.Core/Tutor.Core.csproj

```diff
@@ -6,6 +6,7 @@
 
   <ItemGroup>
     <PackageReference Include="CleanCaDET.CodeModel" Version="1.0.1" />
+    <PackageReference Include="Dahomey.Json" Version="1.12.1" />
```

Comment:
It uses the Dahomey.Json attribute JsonDiscriminator to enable polymorphic serialization and deserialization of events. I agree that this dependency shouldn't exist and that using the Json related attribute for objects in the domain should be avoided, however currently I don't see a really simple way to achieve this. 
One possible solution: Polymorphic serialization in Dahomey.Json works by registering Types in hierarchies and storing them in dictionaries, which are then used to choose which Type to provide to System.Text.Json serializers. The purpose of the JsonDiscriminator attribute is really only to map Types to keys. It might be possible to extend Dahomey.Json (or provide our own implementation similar to theirs) that would directly pass the keys when registering Types instead of using attributes.

Reply:
It would be ideal if we could extract the Dahomey configuration to the serializer class (similar to how we configure EF Core in the TutorContext class). If this is not possible for now we might opt to keep this solution for the time being.

Reply:
Unfortunately I don't think that's possible with Dahomey.Json, as far as I can tell attributes are currently the only way to configure class hierarchies. Maybe we can try looking at other packages that expand System.Text.Json, ideally we'd find one that uses something that looks more like EF Core's fluent API instead of attributes.

Reply:
@Luburic @lukaDoric 
I attempted to resolve this issue in [29db469](https://github.com/Clean-CaDET/tutor/pull/19/commits/29db4698b8fa01d4e4966aeb54bb14f385293652) and [a8ff2ee](https://github.com/Clean-CaDET/tutor/pull/19/commits/a8ff2eea33452f9440d81f0ba36f88907440e410), I would appreciate it if you could take a look at those so we can decide if something like this is acceptable, or if we should look for another package to use instead of Dahomey.Json. The relevant classes are in <code>Infrastructure.Serialization</code>.

Summary of the proposed solution: 
Dahomey.Json's <code>[DefaultDiscriminatorConvention](https://github.com/dahomey-technologies/Dahomey.Json/blob/master/src/Dahomey.Json/Serialization/Conventions/DefaultDiscriminatorConvention.cs)</code> can successfully register a Type only if it has an appropriate <code>JsonDiscriminator</code> attribute, which was the behavior we had issues with. Instead, I made a new implementation of Dahomey.Json's <code>[IDiscriminatorConvention](https://github.com/dahomey-technologies/Dahomey.Json/blob/master/src/Dahomey.Json/Serialization/Conventions/IDiscriminatorConvention.cs)</code> (currently named <code>AllowedTypesDiscriminatorConvention</code>), which keeps a dictionary of Types it's allowed to register and keys to use for them (currently provided through constructors), and works by successfully registering Types that are marked as allowed and rejecting the rest.

Reply:
Current configuration was deemed acceptable at weekly sync.